### PR TITLE
Extend events functionality

### PR DESCRIPTION
**Pull request by Foghrye4**

### CODE REVIEW

**Comment by bonii-xx** [comment]
Why do you need an event to prevent repairing? What's your use case?

File: src/main/java/slimeknights/tconstruct/library/events/TinkerCraftingEvent.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
  */
 @Cancelable
 public class TinkerCraftingEvent extends TinkerEvent {
-  private final ItemStack itemStack;
+  private ItemStack itemStack;
```
Comment (bonii-xx) [inline src/main/java/slimeknights/tconstruct/library/events/TinkerCraftingEvent.java]: No, the events do not allow to change the outcome, this needs to stay immutable.

**Comment by Foghrye4** [comment]
One dude ask me to add Tinker's Construct support to my mod. Actually indeed I don't need those. I could just change NBT tag of an output item instead, correct? Idea of my mod "Iblis" is a item quality system. A more experienced player in creating item, a better item he will craft. See also https://minecraft.curseforge.com/projects/iblis

**Comment by bonii-xx** [comment]
Interesting mod. So your use case would be that the player can't repair a tool he can't craft yet? Like, somebody gives him a manyullyn sword when he's lvl1, so he can't repair it?
In general I'm not convinced that preventing repair causes anything but confusion, preventing the usage of the tool alltogether is probably easier and more constructive.

**Comment by Foghrye4** [comment]
Ok, I will use existing API.

**Pull request closed without merging**